core/src/test/java/org/elasticsearch/action/admin/indices/TransportAnalyzeActionTests.java

```diff
@@ -340,4 +343,54 @@ public void testNormalizerWithIndex() throws IOException {
         assertEquals(1, tokens.size());
         assertEquals("abc", tokens.get(0).getTerm());
     }
+
+    public void testExceedDefaultMaxTokenLimit() throws IOException{
+        // create a string with No. words more than maxTokenCount
+        StringBuilder SbText = new StringBuilder();
+        for (int i = 0; i <= maxTokenCount; i++){
+            SbText.append('a');
+            SbText.append(' ');
+        }
+        String Text = SbText.toString();
+
+        // a request with expain=false
```

Comment:
nit: explain. Also, can you add a short note that this is needed to trigger the simple analysis code path?